lib/revision-selector.jsx

```diff
@@ -1,56 +1,89 @@
-import React from 'react'
+import React, { PropTypes } from 'react'
 import moment from 'moment'
+import { property } from 'lodash';
 
-export default React.createClass( {
+const lastUpdateOf = property( 'data.modificationDate' );
+const revisionSorter = ( a, b ) => lastUpdateOf( a ) - lastUpdateOf( b );
 
-	getDefaultProps: function() {
+const sortedRevisions = revisions => (
+	revisions
+		.slice()
+		.sort( revisionSorter )
+);
+
+export const RevisionSelector = React.createClass( {
+	mixins: [
+		require( 'react-onclickoutside' )
+	],
+
+	getInitialState() {
 		return {
-			revisions: []
+			revisions: sortedRevisions( this.props.revisions ),
+			selection: Infinity,
 		}
 	},
 
-	componentDidMount: function() {
-		// Note: this is intentionally not in state so that the UI doesn't update after resetSelection()
-		this.selection = -1;
+	componentWillReceiveProps( { revisions: nextRevisions } ) {
+		const { revisions: prevRevisions } = this.props;
+
+		if ( nextRevisions === prevRevisions ) {
+			return;
+		}
+
+		this.setState( {
+			revisions: sortedRevisions( nextRevisions ),
+		} );
 	},
 
-	mixins: [
-		require( 'react-onclickoutside' )
-	],
+	componentDidUpdate( { revisions: prevRevisions } ) {
+		const { revisions: nextRevisions } = this.props;
+
+		if ( prevRevisions !== nextRevisions ) {
+			// I'm not sure why exactly, but
+			// this control wasn't refreshing
+			// after loading in the revisions
+			// the first time.
```

Comment:
boy @nfcampos I wish I knew. the rendered HTML ends up having the intended `value` value, but it doesn't render the slider in the right spot. If you clone this and run it, you should be able to comment out this line and see that it ends up incorrect.

also, it's not _after the component mounts_, but _after loading the revisions_. There are typically a few renders that run _before_ the revisions load, since the revisions end up being pulled out of IndexedDB which is asynchronous.